commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -613,3 +613,39 @@ def retire_customer_anonymize_fields(self, customer_id: str, customer_version: i
                          f"with ID: {customer_id}, after LMS retirement with "
                          f"error correlation id {err.correlation_id} and error/s: {err.errors}")
             raise err
+
+    def is_first_time_discount_eligible(self, email: str, code: str) -> bool:
+        """
+        Check if a user is eligible for a first time discount
+        Args:
+            email (str): Email of the user
+            code (str): First time discount code
+        Returns (bool): True if the user is eligible for a first time discount
+        """
+        try:
+            discounted_orders = self.base_client.orders.query(
+                where=[
+                    "customerEmail=:email",
+                    "orderState=:orderState",
+                    "discountCodes(discountCode is defined)"
+                ],
+                predicate_var={'email': email, 'orderState': 'Complete'},
+                expand=["discountCodes[*].discountCode"]
+            )
+
+            if discounted_orders.total < 1:
+                return True
+
+            discounted_orders = discounted_orders.results
+
+            for order in discounted_orders:
+                discount_code = order.discount_codes[0].discount_code.obj.code
```

Comment:
why only `[0]` index?

Reply:
The way expansion for discountCodes[*].discountCode will result in discountCodes still being an array but only having a single entry. Hence I am accessing the only value at 0th index here